common/src/autogluon/common/utils/resource_utils.py

```diff
@@ -159,9 +159,22 @@ def _get_gpu_count_cuda():
         cudaShutdown()
         return gpu_count
 
+    @staticmethod
+    def _get_custom_memory_size():
+        memory_limit = float(os.environ.get("AG_MEMORY_LIMIT_IN_GB"))
```

Comment:
Would it make sense for `TabularPredictor.fit` or `init` to have a `max_memory` argument that sets `os.environ["AG_MEMORY_LIMIT_IN_GB"] = str(max_memory)` internally?

Reply:
Jup, that would be more clean. I will look into it.

Reply:
I just proposed a solution that still fits the PR